ddoc_preprocessor.d

```diff
@@ -194,3 +195,61 @@ auto genHeader(string fileText)
     return updateDdocTag(fileText, ddocKey, newContent);
 }
 
+// parse the menu from the Ddoc file
+auto specTocEntries()
+{
+    alias Entry = Tuple!(string, "name", string, "title", string, "fileName");
+    Entry[] entries;
+
+    static immutable specDir = __FILE_FULL_PATH__.dirName.buildNormalizedPath("spec");
+    static immutable mainFile = specDir.buildPath("./spec.ddoc");
+
+    auto specText = mainFile.readText;
+    if (!specText.findSkip("SUBMENU2"))
+        writeln("Menu file has an invalid format.");
+    foreach (line; specText.splitter("\n"))
+    {
+        enum ddocEntryStart = "$(ROOT_DIR)spec/";
+        if (line.find!(not!isWhite).startsWith(ddocEntryStart))
+        {
+            auto ps = line.splitter(ddocEntryStart).dropOne.front.splitter(",");
+            auto name = ps.front.stripExtension.withExtension(".dd").to!string;
+            auto fileName = specDir.buildPath(name);
+            auto title = ps.dropOne.front.idup.strip;
```

Comment:
I would love to have [`shift`](https://github.com/dlang/phobos/pull/4010) - doing `front` and `popFront` in one step.